dotnet/src/SemanticKernel/SkillDefinition/NativeFunction.cs

```diff
@@ -506,7 +508,7 @@ private static (Func<ITextCompletion?, CompleteRequestSettings?, SKContext, Canc
                     {
                         return parser(value, context.Culture);
                     }
-                    catch (Exception e) when (!e.IsCriticalException())
+                    catch (Exception e)
```

Comment:
By removing the function `IsCriticalException` we are pratically considering any exceptions to be "critical" (should throw).

Two options: 
1. Remove all the catch block
2. Keep the catch just to log and `throw` after